core/domain/feedback_services.py

```diff
@@ -282,15 +291,47 @@ def get_all_threads(exploration_id, has_suggestion):
     return all_threads
 
 
-def enqueue_feedback_message_email_task(user_id):
-    """Adds a 'send feedback email' task into the taskqueue."""
+def get_all_thread_participants(exploration_id, thread_id):
+    """Returns set of all participants of a thread."""
+    return set([m.author_id for m in get_messages(exploration_id, thread_id)])
+
+
+def enqueue_feedback_message_batch_email_task(user_id):
+    """Adds a 'send feedback email' (batch) task into the taskqueue."""
 
     taskqueue_services.enqueue_task(
         feconf.FEEDBACK_MESSAGE_EMAIL_HANDLER_URL, {'user_id': user_id},
         feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS)
 
 
-def enqueue_suggestion_email_task(exploration_id, thread_id):
+def enqueue_feedback_message_instant_email_task(user_id, reference):
+    """Adds a 'send feedback email' (instant) task into the taskqueue."""
+
+    payload = {
+        'user_id': user_id,
+        'reference': reference.to_dict()
+    }
+    taskqueue_services.enqueue_task(
+        feconf.INSTANT_FEEDBACK_EMAIL_HANDLER_URL, payload, 0)
+
+
+def _enqueue_feedback_thread_status_change_email_task(
+        user_id, reference, old_status, new_status):
+    """Adds task for sending email when feedback thread status is changed."""
+
+    payload = {
+        'user_id': user_id,
+        'reference': reference.to_dict(),
+        'updated_status': {
```

Comment:
Now that we have a dedicated handler from this, might as well just make old_status and new_status top-level...